src/tests/ftest/server/multiengine_persocket.py

```diff
@@ -0,0 +1,322 @@
+            "sudo wipefs -a /dev/pmem*",
+            "/usr/bin/ls -l /dev/pmem*",
+            'lsblk|grep -E "NAME|pmem"']
+        for cmd in cleanup_cmds:
+            run_pcmd(self.hostlist_servers, cmd, timeout=90)
+
+    def test_multiengines_per_socket(self):
+        """
+        Test ID: DAOS-12076 Test multiple engines/sockets
+        Test description:
+            (1) Storage prepare --scm-ns-per-socket
+            (2) Start server
+            (3) Start agent
+            (4) Dmg system query
+            (5) Pool create
+            (6) Container create and attributes test
+            (7) IOR test
+            (8) MDTEST
+            (9) Cleanup
+            To launch test:
+            (1) Make sure server is equipped with PMem
+            (2) ./launch.py test_multiengines_per_socket -ts <servers> -tc <agent>
+        :avocado: tags=manual
+        :avocado: tags=hw,medium
+        :avocado: tags=server
+        :avocado: tags=test_multiengines_per_socket
+        """
+        # (1) Storage prepare --scm-ns-per-socket
+        step = 1
+        self.log.info("===(%s)===Storage prepare --scm-ns-per-socket", step)
+        self.daos_server_storage_prepare_reset(step)
+        engines_per_socket = self.params.get(
+            "engines_per_socket", "/run/server_config/*", default=1)
+        self.daos_server_storage_prepare_ns(step, engines_per_socket)
+        cmd = "/usr/bin/ls -l /dev/pmem*"
+        results = run_pcmd(self.hostlist_servers, cmd, timeout=90)
+        retry = 0
+        max_retry = 3
+        while results[0]['exit_status'] != 0 and retry < max_retry:
+            retry += 1
+            self.log.info("===(%s.%s retry)sleep 15 sec, retry server configure "
+                "daos_server_storage_prepare_ns", step, retry)
+            time.sleep(15)
+            cmd = "/usr/bin/ls -l /dev/pmem*"
+            results = run_pcmd(self.hostlist_servers, cmd, timeout=90)
+
+        if retry > max_retry:
+            self.cleanup()
+            self.fail(
+                "#PMem did not show after storage prepare --scm-ns-per-socket and {0} "
+                "retries".format(max_retry))
+
+        # (2) Start server
+        step += 1
+        self.log.info("===(%s)===Start server", step)
+        start_server_cmds = [
+            'lsblk|grep -E "NAME|pmem"',
+            "sudo cp /etc/daos/daos_server.yml_2 /etc/daos/daos_server.yml",
+            "sudo systemctl start daos_server.service"]
+        for cmd in start_server_cmds:
+            results = run_pcmd(self.hostlist_servers, cmd, timeout=90)
+        # Check for server start status
+        if results[0]['exit_status']:
+            self.fail("#Fail on {0}".format(cmd))
+
+        # (3) Start agent
+        step += 1
+        self.log.info("===(%s)===Start agent", step)
+        start_agent_cmds = [
+            "sudo systemctl start daos_agent.service",
+            "dmg storage scan",
+            "dmg network scan",
+            "dmg storage format"]
```

Comment:
Can't we use the `get_dmg_command()` and call `storage_scan()`, `network_scan()` etc?

Reply:
Yes, we can use get_dmg_command() and there is an existing proc to start agent (and server). will be included after reboot included.